homeassistant/components/esphome/number.py

```diff
@@ -0,0 +1,87 @@
+"""Support for esphome sensors."""
+from __future__ import annotations
+
+import math
+
+from aioesphomeapi import NumberInfo, NumberState
+import voluptuous as vol
+
+from homeassistant.components.number import NumberEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+import homeassistant.helpers.config_validation as cv
+
+from . import EsphomeEntity, esphome_state_property, platform_async_setup_entry
+
+ICON_SCHEMA = vol.Schema(cv.icon)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities
+) -> None:
+    """Set up esphome sensors based on a config entry."""
+    await platform_async_setup_entry(
+        hass,
+        entry,
+        async_add_entities,
+        component_key="number",
+        info_type=NumberInfo,
+        entity_type=EsphomeNumber,
+        state_type=NumberState,
+    )
+
+
+# https://github.com/PyCQA/pylint/issues/3150 for all @esphome_state_property
+# pylint: disable=invalid-overridden-method
+
+
+class EsphomeNumber(EsphomeEntity, NumberEntity):
+    """A sensor implementation for esphome."""
+
+    @property
+    def _static_info(self) -> NumberInfo:
+        return super()._static_info
+
+    @property
+    def _state(self) -> NumberState | None:
+        return super()._state
+
+    @property
+    def icon(self) -> str:
+        """Return the icon."""
+        if not self._static_info.icon or self._static_info.device_class:
```

Comment:
```suggestion
        if not self._static_info.icon or self._static_info.device_class:
```

Number static info has no `device_class` attribute as far as I see

Reply:
Removed